dissect/target/plugin.py

```diff
@@ -841,6 +866,43 @@ def find_functions(
     return result, invalid_functions
 
 
+def find_functions_by_record_field_type(
```

Comment:
After some testing using `target-shell`'s python mode on a fresh Linux virtual machine it seems that something along the lines of  `find_functions_by_record_field_type("path", target=t, compatibility=True)` slows things down considerably

The `seen` variable on line 954 in the `_filter_compatible()` function never gets populated. Populating this locally seems to fix the slow down issues.